### PR TITLE
feature 871 TCGen new MET config variables

### LINKED ISSUE #871: Update TCGen config options to use new names
A few MET config variables changed or were added in dtcenter/MET#1750.

## Describe the Enhancement ##
Update TCGen wrapper to set the updated variables.

* Add genesis_match_point_to_track (boolean)
* Add genesis_match_window (dictionary with beg and end integers)
* Remove ops_hit_tdiff and replace it with ops_hit_window (dictionary with beg and end integers)

### Time Estimate ###
~1-2 hours

### Sub-Issues ###
Consider breaking the enhancement down into sub-issues.
- [X] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
4.0.0

### Funding Source ###
*Define the source of funding and account keys here or state NONE.*

### Assignee ###
- [X] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [X] Select **component(s)**
- [X] Select **priority**
- [X] Select **requestor(s)**

### Projects and Milestone ###
- [X] Review **projects** and select relevant **Repository** and **Organization** ones or add "alert:NEED PROJECT ASSIGNMENT" label
- [X] Select **milestone** to next major version milestone or "Future Versions"

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [X] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdatadb](https://github.com/dtcenter/METdatadb/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Enhancement Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)**, **Project(s)**, **Milestone**, and **Linked issues**
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

### PR DESCRIPTION
**Pull request by georgemccabe**
per #871, added support for genesis_match_point_to_track, genesis_match_window{beg;end} and ops_hit_window{beg;end}, removed ops_hit_tdiff

## Pull Request Testing ##

- [X] Describe testing already performed for these changes:</br>
* Added unit tests to ensure METplus config variables set environment variables read by wrapped MET config properly
* Ran TCGen.conf use case using MET from /d1/projects/MET/MET_pull_requests/met-10.0.0_beta5/feature_1714 - once setting all new config variables to default values in TCGenConfig_default and once with new config variables unset. Verified that the same output is created for both.

- [X] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>
Ensure changes work for use case in development that uses TCGen (#630)

- [X] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**

- [X] Do these changes include sufficient testing updates? **[Yes]**

- [X] Will this PR result in changes to the test suite? **[No]**</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [X] Complete the PR definition above.
- [X] Ensure the PR title matches the feature or bugfix branch name.
- [X] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**, **Project(s)**, and **Milestone**
- [x] After submitting the PR, select **Linked Issues** with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

File: parm/use_cases/met_tool_wrapper/TCGen/TCGen.conf [hunk 1]
```diff
@@ -104,13 +104,19 @@ TC_GEN_DLAND_THRESH = NA
...
 TC_GEN_GENESIS_MATCH_RADIUS = 500
 
+#TC_GEN_GENESIS_MATCH_POINT_TO_TRACK = True
+
+#TC_GEN_GENESIS_MATCH_WINDOW_BEG = 0
+#TC_GEN_GENESIS_MATCH_WINDOW_END = 0
+
+#TC_GEN_OPS_HIT_WINDOW_BEG = 0
+#TC_GEN_OPS_HIT_WINDOW_END = 48
+
```
Comment (DanielAdriaansen) [inline parm/use_cases/met_tool_wrapper/TCGen/TCGen.conf:114]: I think these should be uncommented

**Review by DanielAdriaansen — CHANGES_REQUESTED** [review]
See inline comments on TCGen.conf.

File: parm/use_cases/met_tool_wrapper/TCGen/TCGen.conf (hunk 1 above)
Comment (georgemccabe) [reply parm/use_cases/met_tool_wrapper/TCGen/TCGen.conf:114]: These were left commented intentionally. The idea is that the basic met_tool_wrapper use case lists all of the possible config variables that can be used by the wrapper, but do not actually set the optional values. These are here for reference so users can easy see what they can change. If these were uncommented and set to the default value that corresponds to the default values in TCGenConfig_default, then we would have to update this use case conf every time a default value changes on the MET side. I use these variables to test that the values read into the MET config are read properly and don't cause the tool to crash, then comment them out for the final version.

Comment (DanielAdriaansen) [reply parm/use_cases/met_tool_wrapper/TCGen/TCGen.conf:114]: Thanks!

**Review by DanielAdriaansen — APPROVED** [review]
I was able to test these changes and confirm the functionality.

**Pull request merged**